libgeopmd/src/DrmSysfsDriver.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2015 - 2024, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include "DrmSysfsDriver.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <cmath>
+#include <cstring>
+#include <sstream>
+
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm_topo.h"
+
+#include "SysfsIOGroup.hpp"
+
+static const std::string DRM_DIRECTORY = "/sys/class/drm";
+static const std::string GPU_CARD_PREFIX = "card";
+static const std::string GPU_TILE_PREFIX = "gt";
+
+static std::map<int, std::string> load_resources_by_gpu_tile(const std::string &drm_directory, int num_tiles_per_gpu)
+{
+    std::map<int, std::string> result;
+    for (const auto &card_directory : geopm::list_directory_files(drm_directory)) {
+        if (card_directory.find(GPU_CARD_PREFIX) != 0 || card_directory.length() <= GPU_CARD_PREFIX.length()) {
+            continue;
+        }
+        auto card_index = std::stoi(card_directory.c_str() + GPU_CARD_PREFIX.length());
+
+        std::ostringstream card_gt_oss;
+        card_gt_oss << drm_directory << "/" << card_directory << "/gt";
+        for (const auto &tile_directory : geopm::list_directory_files(card_gt_oss.str())) {
+            if (tile_directory.find(GPU_TILE_PREFIX) != 0 || tile_directory.length() <= GPU_TILE_PREFIX.length()) {
+                continue;
+            }
+            auto card_tile_index = std::stoi(tile_directory.c_str() + GPU_TILE_PREFIX.length());
+            if (card_tile_index < 0 || card_tile_index >= num_tiles_per_gpu) {
+                throw geopm::Exception("DrmSysfsDriver encountered an unexpected GPU tile at " + card_gt_oss.str() + "/" + tile_directory,
+                                       GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+            }
+
+            auto global_tile_index = card_index * num_tiles_per_gpu + card_tile_index;
+            result.emplace(global_tile_index, card_gt_oss.str() + "/" + tile_directory);
+        }
+    }
+
+    return result;
+}
+
+namespace geopm
+{
+    const std::string drm_sysfs_json(void);
+
+    DrmSysfsDriver::DrmSysfsDriver()
+        : DrmSysfsDriver(platform_topo(), DRM_DIRECTORY)
+    {
+    }
+
+    DrmSysfsDriver::DrmSysfsDriver(const PlatformTopo &topo,
+                                   const std::string &drm_directory)
+        : M_PROPERTIES{SysfsDriver::parse_properties_json(plugin_name(), drm_sysfs_json())}
+        , M_DRM_RESOURCE_BY_GPU_TILE(load_resources_by_gpu_tile(
+                    drm_directory,
+                    (topo.num_domain(GEOPM_DOMAIN_GPU) > 0) ? (topo.num_domain(GEOPM_DOMAIN_GPU_CHIP) / topo.num_domain(GEOPM_DOMAIN_GPU)) : 0))
```

Comment:
```suggestion
        , M_DRM_RESOURCE_BY_GPU_TILE(load_resources_by_gpu_tile(
                                     drm_directory,
                                     (topo.num_domain(GEOPM_DOMAIN_GPU) > 0) ? 
                                     (topo.num_domain(GEOPM_DOMAIN_GPU_CHIP) /
                                     topo.num_domain(GEOPM_DOMAIN_GPU)) : 0))
```
At least the drm_directory should be aligned under the load_resources... from the previous line.   Not sure if I broke your ternary expression or not.